### DISCUSSION TITLE
qpsolvers v3: Release date and upcoming API changes

### DISCUSSION THREAD

**Discussion by stephane-caron** [qpsolvers/qpsolvers#147] [Announcements]
The next major version of qpsolvers will be released on **February 28, 2023**.

Here is an updated list of the upcoming API changes that will come with this major version increment:

* All open source solvers (that include wheels in their PyPI packages) will be installed by default:
    * https://github.com/stephane-caron/qpsolvers/pull/143
* All exceptions raised by qpsolvers will derive from ``QPError``:
    * https://github.com/stephane-caron/qpsolvers/pull/139
* The ``solve_safer_qp`` function will be removed:
    * https://github.com/stephane-caron/qpsolvers/pull/148
* The ``sym_proj`` parameter will be removed:
    * https://github.com/stephane-caron/qpsolvers/pull/150
* The extra ``__init__.py`` at the root of the repository will be removed
    * https://github.com/qpsolvers/qpsolvers/pull/180

**Comment by wertersilva** [comment]
Does any of the solvers support scipy.sparse.linalg.LinearOperator? Thanks.

**Reply by stephane-caron** [reply]
Moving your question to → https://github.com/stephane-caron/qpsolvers/discussions/172 as it is not related to the release.

**Comment by stephane-caron** [comment]
We're on track to release the next major version of qpsolvers as planned on February 28, 2023. That is, next Tuesday.
